default/stringtables/en.txt

```diff
@@ -12439,24 +12487,39 @@ LRN_DISTRIB_THOUGHT
 Distributed Thought Computing
 
 LRN_DISTRIB_THOUGHT_DESC
-'''Increases [[metertype METER_TARGET_RESEARCH]] on all planets with any focus by 0.1 per [[metertype METER_POPULATION]].
+'''Increases [[metertype METER_TARGET_RESEARCH]] on all planets with any focus by [[value LRN_DISTRIB_THOUGHT_TARGET_RESEARCH_PERPOP]] per [[metertype METER_POPULATION]].
 
 Grants 1 extra military policy slot.
 
 The average citizen literally wastes his mind away, one of the finest computing devices in the universe. By tapping into the idle brain cycles of a sufficiently large population through a network of simple cybernetic transceivers, the research centers of a world can gain access to cheap and plentiful raw computing power.'''
 
+LRN_DISTRIB_THOUGHT_TARGET_RESEARCH_PERPOP
+per population
+
 LRN_STELLAR_TOMOGRAPHY
 Stellar Tomography
 
 LRN_STELLAR_TOMOGRAPHY_DESC
 '''Increases [[metertype METER_TARGET_RESEARCH]] on planets with the Research focus, per [[metertype METER_POPULATION]] and depending on the star type:
-* [[STAR_BLACK]] (x0.2)
-* [[STAR_NEUTRON]] (x0.15)
-* [[STAR_BLUE]] or [[STAR_WHITE]] (x0.1)
-* [[STAR_YELLOW]], [[STAR_ORANGE]] or [[STAR_RED]] (x0.05)
+* [[STAR_BLACK]] (x[[value LRN_STELLAR_TOMO_BLACK_TARGET_RESEARCH_PERPOP]])
+* [[STAR_NEUTRON]] (x[[value LRN_STELLAR_TOMO_NEUTRON_TARGET_RESEARCH_PERPOP]])
+* [[STAR_BLUE]] or [[STAR_WHITE]] (x[[value LRN_STELLAR_TOMO_BRIGHT_TARGET_RESEARCH_PERPOP]])
+* [[STAR_YELLOW]], [[STAR_ORANGE]] or [[STAR_RED]] (x[[value LRN_STELLAR_TOMO_DIME_TARGET_RESEARCH_PERPOP]])
 
 A network of specialized satellites can use tomographic methods to reconstruct the activities occurring deep within a star. These otherwise costly-to-produce conditions are ideal for a variety of research endeavors, with larger benefits for more exotic star types.'''
 
+LRN_STELLAR_TOMO_BLACK_TARGET_RESEARCH_PERPOP
```

Comment:
what are these repeated strings for...?

Reply:
Since Stellar Tomo has different values for each star type group, one named value per star type group is required. And hence, as per the previous comment about how named values work in the stringtables, you need to declare one stringtable entry (macro?) per each referenced named value.

It is indeed quite annoying. I would very much prefer that the mere inclusion of `[[value <named_value_name>]]` was the only thing needed. @agrrr3 ?

Reply:
What is the purpose of these strings? Presumably it's to explain as text what the number that the valueref calculates is for or how it is calculated? Your stringtable entries just contain "per population" which doesn't explain anything.

Reply:
I don't know about the purpose or implementation reasons for the entries like
```
LRN_STELLAR_TOMO_BLACK_TARGET_RESEARCH_PERPOP
per population
```
or 
```
BLD_AUTO_HISTORY_ANALYSER_TARGET_RESEARCH_FLAT
''''''
```
I only know Pedia pages with named values will not work if you don't include these kind of entries in the stringtable.



> Your stringtable entries just contain "per population" which doesn't explain anything.

When parsing named values in the AI code, or when working with FOCS scripts, having that kind of extra info in the named value names allows to differentiate between flat bonuses/maluses (`Value = Value + <Named value>`), population based bonuses7maluses (`Value = Value + Target.TargetPopulation * <Named value>`) and scaling bonuses (`Value = Value * <Named value>`).
The same for indicating the affected meter. By including that info in the named value names, the AI code could parse (e.g.) TargetResearch values including "TARGET_RESEARCH" in their names and automatically be aware of every tech affecting that meter regardless of the FOCS content scripters adjusting the AI python code (assuming content scripters do follow that naming convention).

Reply:
> By including that info in the named value names, the AI code could parse

I'm referring to the "per population" contents of the string, which would normally be localized, not the name (or "key") of the stringtable entry. There are arguments for having a semi-standardized format for naming of valuerefs, but that doesn't have anything to do with the contents of the stringtable.

Reply:
Oh, right. I don't have an informed answer for that.
At first thought writing `''''''` for all those strings since I didn't care about them actually (seem to me like a necessary evil), but I tried to follow @agrrr3 suggestions for the UI. Those "per population" appear there also, although they don't seem necessary there neither since you can see the string names in there so you know if they are per pop. or not.

Reply:
> > By including that info in the named value names, the AI code could parse
> 
> I'm referring to the "per population" contents of the string, which would normally be localized, not the name (or "key") of the stringtable entry. There are arguments for having a semi-standardized format for naming of valuerefs, but that doesn't have anything to do with the contents of the stringtable.

This is a bit conincedental, as the tags did not work without the entries. Then I tried to use the entries for providing description of the calculation and of the intended context ("per population"). Oberlus has a UI suggestion how to better use the context.

If we want to keep this, we should probably use a macro for the stringtable contents  `[[NAMED_VALUE_PER_POP]]` so it works well with localisation

Reply:
> the tags did not work without the entries

Is it easy to understand why?

Reply:
> > the tags did not work without the entries
> 
> Is it easy to understand why?

Not for me. Somewhere a UserString lookup gets performed, but i couldnt pinpoint it. UI/LinkText.cpp LinkTaggedText looked promising as format fits the one i get handed, but it is only used in the pedia panel
```
std::string LinkTaggedText(const std::string& tag, const std::string& stringtable_entry)
{ return "<" + tag + " " + stringtable_entry + ">" + UserString(stringtable_entry) + "</" + tag + ">"; }
```

Reply:
I guess I won't work on introducing more named values until the issue about stringtable cluttering from the unwanted stringtable entries is solved or accepted as a necessary evil.

Reply:
I implemented a workaround to ignore missing stringtable entries for named value refs. See #3206

Reply:
So for me it would be ok not to add the empty entries.

Reply:
I'll remove them then. Less lines in the stringtable files.

Reply:
With the empty strings removed, the lint check fails (referenced keys not found). I guess it will work once #3206 is merged.